### PR TITLE
Add client side validation for register and login form

### LINKED ISSUE #142: Signup form should accept only alphabets in name field
# Expected Behavior

The name field in signup form should only accept alphabets as input

# Current Behavior

The name field accepts both alphabets and numbers as name input

### LINKED ISSUE #164: Implement Client Side Validations for Register and Login
# Expected Behaviour

The Register and Login forms should have Client Side Validation and display appropriate messages for wrong or invalid inputs.

# Current Behaviour

Register Form: The only validation is to Check if both password fields are equal.
Login Form: No Validation, data is directly sent to back end.

**Additionally**

This extends #142 and closing this will solve that too. I am working on a Pull Request for this.

### PR DESCRIPTION
**Pull request by architkshk**
Closes #164
Closes #142

Screenshots - 
Screenshot from 2019-03-13 01-12-19
Screenshot from 2019-03-13 01-12-37
Screenshot from 2019-03-13 01-22-02
Screenshot from 2019-03-13 01-22-09

### CODE REVIEW

**Comment by architkshk** [comment]
@ivantha @root5533 Rebased the commit. Please review.

File: fact-bounty-client/src/pages/login/components/Login.js [hunk 1]
```diff
@@ -175,12 +214,13 @@ class Login extends Component {
...
               color="primary"
               className={this.props.classes.submit}
+              disabled={!this.state.formValid}
             >
               Login
             </Button>
           </form>
           <p>
-            Dont have an account?{" "}
+            Don't have an account?{" "}
```
Comment (kartikpandey2) [inline fact-bounty-client/src/pages/login/components/Login.js]: Wrap it in the template literal string.

File: fact-bounty-client/src/pages/login/components/Login.js [hunk 2]
```diff
@@ -91,8 +94,43 @@ class Login extends Component {
...
+    let { emailValid, passwordValid, errors } = this.state;
+
+    switch (fieldName) {
+      case "email":
+        emailValid = value.match(/^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i);
+        errors.email = emailValid ? "" : "Invalid E-mail";
+        break;
+      case "password":
+        passwordValid = value.length >= 8;
+        errors.password = passwordValid ? "" : "Password too short!";
```
Comment (kartikpandey2) [inline fact-bounty-client/src/pages/login/components/Login.js]: I feel the error message `Password too short!` don't make sense in the login page, Instead, make a general error message which says `Wrong password. Try again or click Forgot password to reset it.`.

**Comment by architkshk** [comment]
@ivantha @root5533 Updated the commit. Please review.

**Comment by root5533** [comment]
Update PR for latest changes to merge.

**Comment by architkshk** [comment]
@root5533 Updated. Please review.

**Pull request merged**